### DISCUSSION TITLE
Consumer consuming files too soon with inotify

### DISCUSSION THREAD

**Discussion by renhardy** [jonaswinkler/paperless-ng#463] [Support]
When importing a document with FTP in the consume directory the consumer starts his process before the file has finished copying and I get the following error:

WARNING Not consuming file /usr/src/paperless/src/../consume/****.pdf.filepart: Unknown file extension.

After which the consumer stops and the final file is not processed.

I am running paperless on docker.

Any way to fix this issue? I have configured it to poll the directory for now, but I would prefer to use the inotify system.

Thank you very much

**Comment by jonaswinkler** [comment]
With inotify, paperless also watches for renamed files. I assume that once your *.pdf.filepart is fully copied, your client moves that to *.pdf. If paperless does not pick up the renamed file, then that's an issue with the file system or the FTP server, and I cannot do anything about that.

**Reply by renhardy** [reply]
Thank you for the reply, I will try to look at the file system and the FTP server.

**Reply by martinpichlo** [reply]
> With inotify, paperless also watches for renamed files. I assume that once your *.pdf.filepart is fully copied, your client moves that to *.pdf. If paperless does not pick up the renamed file, then that's an issue with the file system or the FTP server, and I cannot do anything about that.

I have the same issue, but in a bit different enviroment:
My printer (Canon Maxify MB5150) puts the scans directly (via SMB1) into the consume directory.
The printer is creating the file right at the beginning. But while scanning the file is growing bit by bit.
As result paperless is consuming the file, before the scan process ends.

In my opinion there is now way to setup the mentioned renaming process.
As example I get a pdf with only two pages, instead of all 4 pages (and several errors within the protocol).

Any ideas?

Is there another approach than to wait some minutes within a pre consumption script?

**Comment by krikkert** [comment]
Without coding a "wait until the file size is the same as it was five seconds ago"-delay or an "only consume files that are at least five minutes old"-criterion into paperless, the easiest way is probably to have the printer scan to a staging folder and have a script watching that folder and only moving things into Paperless's consume folder when they're done uploading.
